src/js/yeldaWebchatInjector.js

```diff
@@ -152,68 +156,41 @@ class YeldaChat {
+   * @param {event} e
+   */
   messageListener (e) {
-    if (
-      (e.data === 'closeChat' || e.message === 'closeChat')
-      && document.getElementById('assistant_img') !== null
-    ) {
+    if (document.getElementById('assistant_img') === null) {
+      return false
+    }
+    if (e.data === 'closeChat' || e.message === 'closeChat') {
       document.getElementById('yelda_iframe_container').classList.remove('y_active')
-      setTimeout(function () {
         document.getElementById('assistant_img').style.display = 'block'
         document.getElementById('yelda_iframe_container').style.display = 'none'
-      },
-      1000)
-    } else if (
-      (e.data === 'openChat' || e.message === 'openChat')
-      && document.getElementById('assistant_img') !== null
-    ) {
-      this.triggerEvent(document.getElementById('assistant_img'), 'click')
+    } else if (e.data === 'openChat' || e.message === 'openChat') {
+      this.openChat()
     }
   }
-  /**
-   * handles communcation between parent window and iframe
-  */
-  handleFrameListener () {
-    const eventMethod = window.addEventListener
-      ? 'addEventListener'
-      : 'attachEvent'
-    const eventer = window[eventMethod]
-    const messageEvent = eventMethod === 'attachEvent' ? 'onmessage' : 'message'
-    this.messageListenerBind = this.messageListener.bind(this)
-    eventer(messageEvent, this.messageListenerBind)
-  }
 
   /**
-   * removes the event listener
+   * toggle the event listener
+   * @param {Boolean} is_add
    */
-  removeFrameListener () {
-    const eventMethod = window.addEventListener
-      ? 'removeEventListener'
-      : 'detachEvent'
-    const eventer = window[eventMethod]
-    const messageEvent = eventMethod === 'attachEvent' ? 'onmessage' : 'message'
-    eventer(messageEvent, this.messageListenerBind)
-  }
+  toggleFrameListener (is_add) {
+    let eventMethod
+    const eventCondition = is_add ? 'attachEvent' : 'detachEvent'
 
-  /**
-   * handles window resize event by adding and removing class based on window width
-  */
-  handleOnResize () {
-    if (window.outerWidth < 768) {
-      const w = window
-      const d = document
-      const e = d.documentElement
-      const g = d.getElementsByTagName('body')[0]
-      const x = w.innerWidth || e.clientWidth || g.clientWidth
-      const y = w.innerHeight || e.clientHeight || g.clientHeight
-
-      this.webChatContainer.classList.add('yelda_mobile')
-      this.webChatIframe.style.width = x + 'px'
-      this.webChatIframe.style.height = y + 'px'
+    if (is_add) {
+      eventMethod = window.addEventListener ? 'addEventListener' : 'attachEvent'
     } else {
-      this.webChatContainer.classList.remove('yelda_mobile')
+      eventMethod = window.removeEventListener ? 'removeEventListener' : 'detachEvent'
     }
+
+    const eventer = window[eventMethod]
+    const messageEvent = eventMethod === eventCondition ? 'onmessage' : 'message'
```

Comment:
Thanks for handling this case, but could be simplified (like removing eventCondition)

```javascript
// attachEvent is a non-standard JS function implemented in IE8 and previous versions
//  addEventListener is supported by IE9+ (and all the other browsers).
let eventMethod =  window.addEventListener ? 'addEventListener' : 'attachEvent'
const messageEvent = window.addEventListener ? 'onmessage' : 'message'
 
if (!isAdded) {
  eventMethod = window.removeEventListener ? 'removeEventListener' : 'detachEvent'
}

this.messageListenerBind = this.messageListener.bind(this)
window[eventMethod](messageEvent, this.messageListenerBind)
```